### ISSUE TITLE
TFchain client test suite is broken

### ISSUE THREAD

**Issue by zaibon** [threefoldtecharchive/jumpscaleX_archived#563]
JSX revision: b1b16a20bc7d16aef87c65f2f86f19c1b067ed2a

I've try to run the tfchain client tests suite and failed.
Here is a list of all the test that failed:

- atomicswap_initiate
- atomicswap_participate
- atomicswap_verify_sender
- atomicswap_verify_receiver
- wallet_coins_send

All the atomic_swap related test failed because of 
```
Traceback (most recent call last):
  File "<stdin>", line 6, in <module>
  File "/sandbox/lib/jumpscale/Jumpscale/clients/blockchain/tfchain/TFChainClientFactory.py", line 44, in test
    self._test_run(name=name)
  File "/sandbox/lib/jumpscale/Jumpscale/core/BASECLASSES/JSBase.py", line 471, in _test_run
    res = self.__test_run(name=name, obj_key=obj_key, die=die, **kwargs)
  File "/sandbox/lib/jumpscale/Jumpscale/core/BASECLASSES/JSBase.py", line 507, in __test_run
    self.__test_run(name=bname, obj_key=obj_key, **kwargs)
  File "/sandbox/lib/jumpscale/Jumpscale/core/BASECLASSES/JSBase.py", line 528, in __test_run
    res = method(self=self, **kwargs)
  File "/sandbox/lib/jumpscale/Jumpscale/clients/blockchain/tfchain/tests/13_atomicswap_verify_receiver.py", line 81, in main
    w.atomicswap.verify("dd1babcbab492c742983b887a7408742ad0054ec8586541dd6ee6202877cb486", min_refund_time="+1d12h")
  File "/sandbox/lib/jumpscale/Jumpscale/clients/blockchain/tfchain/TFChainWallet.py", line 984, in verify
    min_refund_time = OutputLock(value=min_refund_time, current_timestamp=chain_time).value
  File "/sandbox/lib/jumpscale/Jumpscale/clients/blockchain/tfchain/types/ConditionTypes.py", line 279, in __init__
    offset = j.data.types.duration.fromString(value[1:])
AttributeError: 'Types' object has no attribute 'duration'
```

I created an issue specially for this: https://github.com/threefoldtech/jumpscaleX/issues/564

The test `wallet_coin_send` failed with:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/sandbox/lib/jumpscale/Jumpscale/clients/blockchain/tfchain/TFChainClientFactory.py", line 44, in test
    self._test_run(name=name)
  File "/sandbox/lib/jumpscale/Jumpscale/core/BASECLASSES/JSBase.py", line 471, in _test_run
    res = self.__test_run(name=name, obj_key=obj_key, die=die, **kwargs)
  File "/sandbox/lib/jumpscale/Jumpscale/core/BASECLASSES/JSBase.py", line 507, in __test_run
    self.__test_run(name=bname, obj_key=obj_key, **kwargs)
  File "/sandbox/lib/jumpscale/Jumpscale/core/BASECLASSES/JSBase.py", line 528, in __test_run
    res = method(self=self, **kwargs)
  File "/sandbox/lib/jumpscale/Jumpscale/clients/blockchain/tfchain/tests/24_wallet_coins_send.py", line 167, in main
    assert result.transaction.json() == expected_transaction
```

**Comment by zaibon** [comment]
Error related to Duration typo missing is fixed in https://github.com/threefoldtech/jumpscaleX/issues/564

**Comment by zaibon** [comment]
fixed in f234be9df64282b828d683f243c10d9207a59ef1
